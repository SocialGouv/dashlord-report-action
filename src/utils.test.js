// @ts-nocheck

const fs = require("fs");
const { toHostname, getUrls } = require("./utils");

jest.mock("fs");

const tests = [
  ["http://www.free.fr", "www.free.fr"],
  ["http://sub.www.free.fr", "sub.www.free.fr"],
  ["http://free.fr/some/page", "free.fr"],
  ["free.fr/some/page", "free.fr"],
  ["/some/page", null],
];

describe("toHostname", () => {
  tests.forEach(([input, output]) => {
    test(`should extract ${output} in ${input}`, () => {
      //@ts-ignore
      expect(toHostname(input)).toEqual(output);
    });
  });
});

const sampleConfig = `
title: Dashboard title
urls:
  - url: https://www.free.fr
    title: Homepage free.fr
    tags:
      - telecom
      - provider
  - url: https://www.voila.fr
    title: something
    tags:
      - oldie
`;

beforeEach(() => {
  jest.resetAllMocks();
});

describe("getUrls", () => {
  test("should parse urls.txt files correctly", () => {
    fs.existsSync
      //@ts-expect-error
      .mockImplementationOnce(() => false) // yaml
      .mockImplementationOnce(() => false) // yml
      .mockImplementationOnce(() => true); // txt

    fs.readFileSync.mockReturnValueOnce(`

url1
url2
# comment
url3

`);

    expect(getUrls()).toEqual([
      { url: "url1" },
      { url: "url2" },
      { url: "url3" },
    ]);
  });

  test("should parse dashlord.yaml file correctly", () => {
    fs.existsSync
      //@ts-expect-error
      .mockImplementationOnce(() => true) // yaml
      .mockImplementationOnce(() => false) // yml
      .mockImplementationOnce(() => false); //txt

    fs.readFileSync.mockReturnValueOnce(sampleConfig);

    expect(getUrls()).toMatchSnapshot();
  });

  test("should parse dashlord.yml file correctly", () => {
    fs.existsSync
      //@ts-expect-error
      .mockImplementationOnce(() => false) // yaml
      .mockImplementationOnce(() => true) // yml
      .mockImplementationOnce(() => false); //txt

    fs.readFileSync.mockReturnValueOnce(sampleConfig);

    expect(getUrls()).toMatchSnapshot();
  });
});
