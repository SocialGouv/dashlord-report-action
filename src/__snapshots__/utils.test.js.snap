// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUrls should parse dashlord.yaml file correctly 1`] = `
Array [
  Object {
    "tags": Array [
      "telecom",
      "provider",
    ],
    "title": "Homepage free.fr",
    "url": "https://www.free.fr",
  },
  Object {
    "tags": Array [
      "oldie",
    ],
    "title": "something",
    "url": "https://www.voila.fr",
  },
]
`;

exports[`getUrls should parse dashlord.yml file correctly 1`] = `
Array [
  Object {
    "tags": Array [
      "telecom",
      "provider",
    ],
    "title": "Homepage free.fr",
    "url": "https://www.free.fr",
  },
  Object {
    "tags": Array [
      "oldie",
    ],
    "title": "something",
    "url": "https://www.voila.fr",
  },
]
`;
